planetiler-core/src/main/java/com/onthegomap/planetiler/reader/GeoPackageReader.java

```diff
@@ -138,7 +148,21 @@ public void readFeatures(Consumer<SimpleFeature> next) throws Exception {
       MathTransform transform = (coordinateTransform != null) ? coordinateTransform :
         CRS.findMathTransform(CRS.decode("EPSG:" + srsId), latLonCRS);
 
-      for (var feature : features.queryForAll()) {
+      FeatureIndexManager indexer = new FeatureIndexManager(geoPackage,
+        features);
+
+      Iterable<FeatureRow> results;
+
+      if (this.bounds != null && indexer.isIndexed() && srsId == 4326) {
```

Comment:
Another option is to transform the bounds from 4326 to whatever CRS the geopackage source uses. See ShapefileReader as an example

Reply:
I wasn't sure if that handled non-conformal translations correctly, but if so we can use that. How were you able to use the Shapefile filter without `.sbx` sidecar files?

Reply:
Geotools `ReferencedEnvelope#transform` drops a bunch of points along the edge of the bounding box in the old coordinate system, then computes the bounding box of those points in the new coordinate system. Also if it throws a TransformException then it just skips the filter.

I'm not 100% sure how the geotools filter predicate works internally. My guess/hope would be that it uses the spatial index if available, but then falls back to just applying it against each shape if not. I added it based off the description here: https://docs.geotools.org/latest/userguide/library/main/filter.html